builder/amazon/common/step_security_group.go

```diff
@@ -45,7 +45,9 @@ func (s *StepSecurityGroup) Run(state multistep.StateBag) multistep.StepAction {
 
 	port := s.CommConfig.Port()
 	if port == 0 {
-		panic("port must be set to a non-zero value.")
+		if s.CommConfig.Type != "none" {
+			panic("port must be set to a non-zero value.")
```

Comment:
I don't think this is correct. What port is it going to create a security group for when the communicator is none?

I think we should skip this step entirely when using the nil (none) communicator. I think the existence of the panic line means this code is buggy. It should either not get here at all, or be caught in prepare